pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -250,9 +289,25 @@ spec:
                       description: Tags is a map of key/value tags used to select
                         subnets Specifying '*' for a value selects all values for
                         a given tag key.
+                      maxProperties: 20
                       type: object
+                      x-kubernetes-validations:
+                      - message: empty tag keys or values aren't supported
+                        rule: self.all(k, k != '' && self[k] != '')
                   type: object
+                maxItems: 30
                 type: array
+                x-kubernetes-validations:
+                - message: securityGroupSelectorTerms cannot be empty
+                  rule: self.size() != 0
+                - message: must contain tags, id, or name
```

Comment:
This is different from the style of the messages in the `amiSelectorTerms`. Can we make sure that we are consistent with the messages that we are using 

```
- message: expected at least one, got none, ['tags', 'id', 'name']
```